l10n_es_aeat_mod349/models/mod349.py

```diff
@@ -0,0 +1,460 @@
+        self.total_partner_records = len(self.partner_record_ids)
+        self.total_partner_records_amount = sum([
+            record.total_operation_amount for record in
+            self.partner_record_ids])
+        self.total_partner_refunds = len(self.partner_refund_ids)
+        self.total_partner_refunds_amount = sum([
+            refund.total_operation_amount for refund in
+            self.partner_refund_ids])
+
+    @api.one
+    @api.depends()
+    def _get_report_alias(self):
+        """Returns an alias as name for the report."""
+        self.name = '%s - %s/%s' % (
+            self.company_id.name or '', self.fiscalyear_id.name or '',
+            self.period_selection or '')
+
+    def _create_349_partner_records(self, invoices, partner, operation_key):
+        """creates partner records in 349"""
+        rec_obj = self.env['l10n.es.aeat.mod349.partner_record']
+        partner_country = partner.country_id
+        sum_credit = sum([invoice.cc_amount_untaxed for invoice in invoices
+                          if invoice.type not in ('in_refund', 'out_refund')])
+        sum_debit = sum([invoice.cc_amount_untaxed for invoice in invoices
+                         if invoice.type in ('in_refund', 'out_refund')])
+        invoice_created = rec_obj.create(
+            {'report_id': self.id,
+             'partner_id': partner.id,
+             'partner_vat': _format_partner_vat(partner_vat=partner.vat,
+                                                country=partner_country),
+             'operation_key': operation_key,
+             'country_id': partner_country.id or False,
+             'total_operation_amount': sum_credit - sum_debit
+             })
+        # Creation of partner detail lines
+        for invoice in invoices:
+            detail_obj = self.env['l10n.es.aeat.mod349.partner_record_detail']
+            detail_obj.create({'partner_record_id': invoice_created.id,
+                               'invoice_id': invoice.id,
+                               'amount_untaxed': invoice.cc_amount_untaxed})
+        return invoice_created
+
+    def _create_349_refund_records(self, refunds, partner, operation_key):
+        """Creates restitution records in 349"""
+        partner_detail_obj = self.env[
+            'l10n.es.aeat.mod349.partner_record_detail']
+        obj = self.env['l10n.es.aeat.mod349.partner_refund']
+        obj_detail = self.env['l10n.es.aeat.mod349.partner_refund_detail']
+        partner_country = partner.country_id
+        record = {}
+        for refund in refunds:
+            # goes around all refunded invoices
+            for origin_inv in refund.origin_invoices_ids:
+                if origin_inv.state in ('open', 'paid'):
+                    # searches for details of another 349s to restore
+                    refund_details = partner_detail_obj.search(
+                        [('invoice_id', '=', origin_inv.id),
+                         ('report_id', '!=', False)])
```

Comment:
ValueError: Invalid field 'report_id' in leaf "<osv.ExtendedLeaf: ('report_id', '!=', False) on l10n_es_aeat_mod349_partner_record_detail (ctx: )>"
report_id no existe y tampoco existía en la anterior versión, así que no tengo muy claro el propósito de esa condición, pero creo que se puede eliminar sin más